cron/hourly/20-fix-stripe-stale-NEW-paymentIntents.ts

```diff
@@ -0,0 +1,85 @@
+import '../../server/env';
+
+import { isEmpty } from 'lodash';
+import moment from 'moment';
+
+import OrderStatuses from '../../server/constants/order-status';
+import { PAYMENT_METHOD_SERVICE, PAYMENT_METHOD_TYPE } from '../../server/constants/paymentMethods';
+import logger from '../../server/lib/logger';
+import { syncOrder } from '../../server/lib/stripe/sync-order';
+import { Collective, Op, Order, PaymentMethod } from '../../server/models';
+import { runCronJob } from '../utils';
+
+const PAGE_SIZE = 20;
+
+async function* staleStripeNewPaymentIntentOrdersPager() {
+  const query = {
+    include: [
+      {
+        model: PaymentMethod,
+        as: 'paymentMethod',
+        where: {
+          service: PAYMENT_METHOD_SERVICE.STRIPE,
+          type: PAYMENT_METHOD_TYPE.PAYMENT_INTENT,
+        },
+        required: true,
+      },
+      { model: Collective, as: 'collective' },
+    ],
+    where: {
+      status: OrderStatuses.NEW,
+      data: {
+        paymentIntent: {
+          id: {
+            [Op.ne]: null,
+          },
+        },
+      },
+      createdAt: {
+        [Op.lte]: moment().subtract(2, 'days').toDate(),
+      },
+    },
+  };
+
+  const total = await Order.count(query);
+
+  logger.info(`${total} stale (>=2 days old) stripe payment intent orders with status NEW.`);
+
+  if (total === 0) {
+    return;
+  }
+
+  let offset = 0;
+  while (true) {
+    const pageResult = await Order.findAll({
+      ...query,
+      limit: PAGE_SIZE,
+      offset,
+    });
+
+    if (isEmpty(pageResult)) {
+      return;
+    }
+
+    yield pageResult;
+
+    offset += PAGE_SIZE;
+  }
```

Comment:
This is not a safe way to paginate orders as fewer or more results will end up changing the results.

Example - take 5 orders:
- order 1
- order 2
- order 3
- order 4
- order 5

And say the pagination is "3". If nothing changes, you'll get [1,2,3] then [4,5]. So far so good.

Now imagine a new order "0" matches the conditions while you're processing order 1 (e.g. because it's `createdAt` just reached the condition threshold):
- order 0
- order 1
- order 2
- order 3
- order 4
- order 5

While your first page will be [1,2,3], the second query will now return [3,4,5] resulting in a double-processing of `3`.

Similar things can happen if orders get removed from the query, and given that `syncOrder` updates the status it will impact pagination.